datafusion/core/src/physical_plan/file_format/mod.rs

```diff
@@ -541,6 +541,23 @@ pub(crate) fn get_output_ordering(
         }).unwrap_or_else(|| None)
 }
 
+/// Defines strategies for the saving behavior in the case when the saved folder already exists
+#[derive(Debug, Clone, Copy)]
+pub enum FileWriterSaveMode {
+    /// Overwrite mode means that when saving a parquet file, if folder already exists, existing data is expected to be overwritten.
+    /// If folder does not exist, the folder will be created. This is default value.
+    Overwrite,
+    /// Append mode means that when saving a parquet file, if folder already exists, new data is expected to be appended on top of the existing.
+    /// If folder does not exist, the folder will be created
+    Append,
```

Comment:
As I understand, parquet files are not appended, we hold the directory and write new files with a new uuid. 

This approach is specific to the parquet file format and may not apply to other file formats like CSV, where appending to an existing file is the expected behavior. It's worth noting that this structure may be extended to support other file formats in the future, so it's important to update the comments accordingly.

Reply:
I see your point, will be trying to do writer properties more sbstract, so each writer can implement its own save mode behaviuors